test/testwindow.go

```diff
@@ -37,6 +38,11 @@ func (w *testWindow) Clipboard() fyne.Clipboard {
 }
 
 func (w *testWindow) Close() {
+	if w.onCloseIntercepted != nil {
```

Comment:
To match the pattern this should not really be here. Perhaps the test window might need a `simulateUserClose()` or similar function that would test the intercept.
Either that or I was wrong about this being testable here - maybe test windows always close when requested?